aten/src/ATen/CMakeLists.txt

```diff
@@ -117,6 +117,7 @@ file(GLOB native_cpu_h "native/cpu/*.h")
 
 file(GLOB native_cuda_cu "native/cuda/*.cu")
 file(GLOB native_cuda_cpp "native/cuda/*.cpp")
+file(GLOB native_grid_sampler_utils_cpp "native/GridSamplerUtils.cpp")
```

Comment:
These CMake changes shouldn't be necessary. Just add `TORCH_API` to each function in `GridSamplerUtils.h`.

Reply:
i'll give it a try, but i did this because it failed during linking. iiuc, usually people put cuda code under `native/cuda` , but this one is one level above to allow sharing between cpu and cuda. it would work as is if it were a header-only change (you'd need to put the functions into an anonymous namespace tho to avoid multiple definition errors)

Reply:
That's exactly what `TORCH_API` does; it exports the symbol from `torch_cpu.so` so it can be used in `torch_cuda.so` and is linked correctly. There is no need to compile it twice.

Reply:
looks like this approach works, will push as part of the next update